src/routes/privateSnotra.ts

```diff
@@ -0,0 +1,60 @@
+import { FastifyInstance } from 'fastify';
+import createOrGetConnection from '../db';
+import { Keyword, Source, Post } from '../entity';
+
+export default async function (fastify: FastifyInstance): Promise<void> {
+  fastify.get('/all_tags', async (req, res) => {
+    if (!req.service) {
+      return res.status(400).send();
+    }
+
+    const con = await createOrGetConnection();
+    const result = await con
+      .createQueryBuilder()
+      .select('value', 'name')
+      .from(Keyword, 'keyword')
+      .where('status = :status', { status: 'allow' })
+      .orderBy('name', 'ASC')
+      .getRawMany<{ name: string }>();
+
+    return res.status(200).send(result);
+  });
+
+  fastify.get('/all_sources', async (req, res) => {
+    if (!req.service) {
+      return res.status(400).send();
+    }
+
+    const con = await createOrGetConnection();
+    const result = await con
+      .createQueryBuilder()
+      .select('id', 'name')
+      .from(Source, 'source')
+      .where('type = :type', { type: 'machine' })
+      .andWhere('private = :private', { private: false })
+      .andWhere('active = :active', { active: true })
+      .orderBy('name', 'ASC')
```

Comment:
Do we order by name or id in BQ?

Reply:
```
SELECT * FROM EXTERNAL_QUERY("us.api","SELECT CONCAT('source_', LOWER(id)) AS name FROM source WHERE type = 'machine' AND private = false AND active")
UNION ALL
SELECT * FROM EXTERNAL_QUERY('us.api',"SELECT CONCAT('tag_', LOWER(value)) AS name from keyword where status = 'allow'")
ORDER BY 1
```
This is an original query, it uses a different approach: takes all tags and sources and orders them together later. I added ordering to receive consistent responses, but it is not strongly necessary.

Reply:
Don't get me wrong, I think the ordering is needed but I'd suggest using id. Name can be changed, id can't

Reply:
@idoshamun aah, ok, I got what you mean now. Sorry, I got confused xD. In the sources table, we have columns "id" and "name", the name can be changed, id can not. I am querying ID as NAME, so it is should be fine :)

Reply:
Ok then we can merge!